sycl/test-e2e/format.py

```diff
@@ -154,18 +195,26 @@ def execute(self, test, litConfig):
         if isinstance(script, lit.Test.Result):
             return script
 
-        devices_for_test = self.select_devices_for_test(test)
-        if not devices_for_test:
-            return lit.Test.Result(
-                lit.Test.UNSUPPORTED, "No supported devices to run the test on"
-            )
-
-        substitutions = lit.TestRunner.getDefaultSubstitutions(test, tmpDir, tmpBase)
+        devices_for_test = []
         triples = set()
-        for sycl_device in devices_for_test:
-            (backend, _) = sycl_device.split(":")
-            triples.add(get_triple(test, backend))
+        if "run-mode" not in test.config.available_features:
```

Comment:
maybe swap the logic so the first check isnt a `not`? could we check for `build` instead?

Reply:
The issue with doing this is that the non-split mode sets both run-mode and build-mode features. what i am doing here is specifically checking for the split build *only* mode.

Reply:
ah ok thx
build and run mode at the same time just does exactly what we do today without these changes right?

Reply:
> ah ok thx build and run mode at the same time just does exactly what we do today without these changes right?

Correct.

Reply:
I haven't looked through the PR yet, but from previous discussions, I'd expect that we might build more tests/for more triples that are otherwise skipped in the old mode due to XFAIL/UNSUPPORTED.

Reply:
you mean for tests that compile fine but fail/unsupported at runtime, we always compile those but don't run the test? yeah that sounds cool